### PR TITLE
Update repo with utilities/components from Make a Gift; Use Tailwind JIT compiler

### PR DESCRIPTION
**Pull request by yvonnetangsu**
# NOT READY FOR REVIEW
- (Edit the above to reflect status)

# Summary
- TL;DR - what's this PR for?

# Review By (Date)
- When does this need to be reviewed by?

# Criticality
- How critical is this PR on a 1-10 scale? Also see [Severity Assessment](https://stanfordits.atlassian.net/browse/D8CORE-1705).
- E.g., it affects one site, or every site and product?

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Navigate to...
3. Verify...

## Front End Validation
- [ ] Is the markup using the appropriate semantic tags and passes HTML validation?
- [ ] Cross-browser testing has been performed?
- [ ] Automated accessibility scans performed?
- [ ] Manual accessibility tests performed?
- [ ] Design is approved by @ user?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
- JIRA ticket(s)
- Other PRs
- Any other contextual information that might be helpful (e.g., description of a bug that this PR fixes, new functionality that it adds, etc.)
- Anyone who should be notified? (`@mention` them here)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/**
```
Comment (sherakama) [inline tailwind.config.js:1]: Is it possible to pull in the settings file in Decanter and then lay the custom theme stuff on top?

File: tailwind.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+var dir = path.resolve(__dirname, 'src/tailwind/plugins')
+
+module.exports = {
+  mode: 'jit',
+  prefix: 'su-',
+
+  purge: [
+    './public/**/*.html',
+    './src/**/*.{js,jsx,ts,tsx,vue}',
+  ],
```
Comment (sherakama) [inline tailwind.config.js:15]: Is purge needed with the JIT compiler?

File: tailwind.config.js (hunk 1 above)
Comment (yvonnetangsu) [reply tailwind.config.js:1]: Yeah, that's kind of what I'm hoping - like the SAA stuff would be extending the Decanter stuff. Haven't had a lot of time to spend on this yet - will continue testing.

File: tailwind.config.js (hunk 2 above)
Comment (yvonnetangsu) [reply tailwind.config.js:15]: Apparently it still needed so it knows whether to generate the class:
https://github.com/tailwindlabs/tailwindcss-jit

**Comment by sherakama** [comment]
> Looks like this is working!

Ooo yay!

File: tailwind.config.js (hunk 2 above)
Comment (yvonnetangsu) [reply tailwind.config.js:20]: > > Looks like this is working!
> 
> Ooo yay!

Works in development mode, but looks like production mode is stripping out the Decanter stuff 🤔

**Comment by yvonnetangsu** [comment]
This is what it looks like when I do `npm run develop` - which is exactly what I want. It's generating the classes from the Decanter custom stuff plus the classes generated by the SAA extension to the theme

Screen Shot 2021-03-30 at 1 53 00 PM

But things are missing when I do `npm run build` 😞

**Comment by yvonnetangsu** [comment]
@sherakama So it looks like `npm run build` actually generated all the styles correctly also - this is when I clicked on the html file in the public folder. Looks as expected - same for the new 404 and 403 pages that I just published. But for some reason, the netlify preview aren't the same. I already tried the "clean cache and deploy" option 🤔 

Screen Shot 2021-03-30 at 3 59 39 PM

**Comment by sherakama** [comment]
I'll have a look a little later. Looks like there might be something still up with the dotenv stuff.

**Comment by yvonnetangsu** [comment]
Thanks! And here are some warning messages from npm run build

Screen Shot 2021-03-30 at 4 31 54 PM

File: tailwind.config.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    require(decanter + '/components/tables/borderless.js')(),
+    require(decanter + '/components/typography/modular-typography.js')(),
+    require(decanter + '/components/typography/styles.js')(),
+    require(decanter + '/components/typography/wysiwyg.js')(),
+
+    // @tailwind utilities;
+    require(decanter + '/utilities/accessibility/accessibility-hidden.js')(),
+    require(decanter + '/utilities/link/link.js')(),
+    require(decanter + '/utilities/link/link-fontweight.js')(),
+    require(decanter + '/utilities/link/link-underline.js')(),
```
Comment (yvonnetangsu) [inline tailwind.config.js:67]: Found one more issue - I think by pulling these ones (those that we custom built the variants for) in from Decanter this way, the hover/focus variants don't work quite right. The other ones seem ok though.

**Comment by yvonnetangsu** [comment]
@sherakama I know you're out Friday and next Monday, could I bother you to take a look at this? 😃  It works in dev mode and production mode locally, just not on Netlify. If it's going to take a long time then let's wait till next week. I'm not really blocked since I can work on local footer and preview locally. Thanks!

**Comment by sherakama** [comment]
I'll find some time after dev chat today.

**Comment by yvonnetangsu** [comment]
Thanks! And like I said, I can still work on the other components, so no big deal if you have other things on your plate (which I'm sure you do).

**Comment by yvonnetangsu** [comment]
Closing this one in favor of the other PRs that branch off this one - avoid confusion

**Comment by yvonnetangsu** [comment]
Please see #6 and #8 instead - things added in this branch are in those 2 branches as well

**Pull request closed without merging**